### PR TITLE
Create user crud

### PR DESCRIPTION
**Pull request by jdhiguita**
## Summary
- Create CRUD for user controller
- Create add and remove book
# Trello Card
https://trello.com/c/4Pmk2Oq9

### CODE REVIEW

File: src/main/java/wolox/training/services/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import wolox.training.services.IUserService;
+
+@Service
+public class UserServiceImpl implements IUserService {
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    BookRepository bookRepository;
```
Comment (cramallo) [inline src/main/java/wolox/training/services/impl/UserServiceImpl.java]: You could inject the service here

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -45,6 +47,9 @@
     @Column(nullable = false, unique = true)
     private String isbn;
 
+    @ManyToMany(mappedBy = "books")
+    private List<User> users;
+
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java:53]: Add a get with Collections.unmodifiableList for users list.

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/services/impl/UserServiceImpl.java (hunk 1 above)
Comment (jdhiguita) [reply src/main/java/wolox/training/services/impl/UserServiceImpl.java]: Update repository by service

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (jdhiguita) [reply src/main/java/wolox/training/models/Book.java:53]: Add Collections.unmodifiableList in the get of users list.

**Review by Gianni-Nicolas — APPROVED** [review]

**Review by cramallo — APPROVED** [review]

**Pull request merged**